### PR TITLE
Task service - basic infrastructure to register and run tasks using machinery

### PR DESCRIPTION
**Pull request by yegorch**
* Add dependencies for machinery
* Task service basic impl

### CODE REVIEW

**Comment by saheienko** [comment]
```
 vendor/github.com/RichardKnop/machinery/vendor/github.com/RichardKnop/logging/interface.go
```
Vendoring a vendor directory?

**Comment by yegorch** [comment]
@saheienko it's normal govendor behaviour https://github.com/kardianos/govendor/issues/247

File: pkg/profile/kube_profile.go [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-	KubernetesVersion string        `json:"kubernetes_version" valid:"required"`
-	Provider          string        `json:"provider" valid:"in(aws|digitalocean|packet|gce|openstack)"`
-	Nodes             []NodeProfile `json:"nodes" valid:"required"`
-	MasterNodeCount   int           `json:"master_node_count" valid:"required"`
-	CustomFiles       string        `json:"custom_files" valid:"optional"`
-	RBACEnabled       bool          `json:"rbac_enabled"`
+	Id                string      `json:"id" valid:"required"`
+	KubernetesVersion string      `json:"kubernetes_version" valid:"required"`
+	MasterProfile     NodeProfile `json:"master_profile" valid:"required"`
+	NodesProfile      NodeProfile `json:"nodes_profile" valid:"required"`
```
Comment (saheienko) [inline pkg/profile/kube_profile.go]: Please, use masterProfile instead of master_profile as a json tag.

File: pkg/profile/node_profile.go [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
+
 type NodeProfile struct {
-	Id       string `json:"id" valid:"required"`
-	Size     string `json:"size" valid:"required"`
-	Image    string `json:"image" valid:"required"`
-	Provider string `json:"provider" valid:"in(aws|digitalocean|packet|gce|openstack)"`
+	Id       string        `json:"id" valid:"required"`
+	Size     string        `json:"size" valid:"required"`
+	Image    string        `json:"image" valid:"required"`
+	Provider provider.Name `json:"provider" valid:"in(aws|digitalocean|packet|gce|openstack)"`
```
Comment (saheienko) [inline pkg/profile/node_profile.go]: It is a bad idea to check a list of providers here. If we add a new one, we should remember to update all the tags.

File: pkg/account/service.go [hunk 3]
```diff
@@ -69,7 +70,7 @@ func (s *Service) Get(ctx context.Context, accountName string) (*CloudAccount, e
 func (s *Service) Create(ctx context.Context, account *CloudAccount) error {
 	logrus.Debug("cloud_account.Service.Create start")
 
-	rawJSON, err := json.Marshal(account)
+	rawJSON, err := json	.Marshal(account)
```
Comment (saheienko) [inline pkg/account/service.go]:
```
rawJSON, err := json	.Marshal(account)
```

File: pkg/profile/node_profile.go [hunk 4]
```diff
@@ -1,10 +1,20 @@
...
 	// AVX256, SSE4, MMX, AES, SR-IOV etc.
 	Capabilities []string `json:"capabilities" valid:"optional"`
 }
+
+type NodeSize struct {
+	CPU string `json:"cpu"`
+	RAM string `json:"ram"`
+
+	// Name e.g. t2.micro
+	Name string `json:"slur"`
```
Comment (saheienko) [inline pkg/profile/node_profile.go]:
```
Name string `json:"slur"`
```
Is it valid - "slur"?

**Pull request merged**